completion/waffle.py

```diff
@@ -21,9 +23,67 @@
 # xblocks.
 ENABLE_COMPLETION_TRACKING = 'enable_completion_tracking'
 
+# Full name completion.enable_visual_progress
+# Overrides completion.enable_course_visual_progress
+# Acts as a global override -- enable visual progress indicators
+# sitewide.
+ENABLE_VISUAL_PROGRESS = 'enable_visual_progress'
+
+# Full name completion.enable_course_visual_progress
+# Acts as a course-by-course enabling of visual progress
+# indicators, e.g. updated 'resume button' functionality
+ENABLE_COURSE_VISUAL_PROGRESS = 'enable_course_visual_progress'
+
+# SiteConfiguration visual progress enablement
+ENABLE_SITE_VISUAL_PROGRESS = 'enable_site_visual_progress'
+
 
 def waffle():
     """
     Returns the namespaced, cached, audited Waffle class for completion.
     """
     return WaffleSwitchNamespace(name=WAFFLE_NAMESPACE, log_prefix='completion: ')
+
+
+def waffle_flag():
+    """
+    Returns the namespaced, cached, audited Waffle flags dictionary for Completion.
+    """
+    namespace = WaffleFlagNamespace(name=WAFFLE_NAMESPACE, log_prefix=u'completion: ')
+    return {
+        # By default, disable visual progress. Can be enabled on a course-by-course basis.
+        # And overridden site-globally by ENABLE_VISUAL_PROGRESS
+        ENABLE_COURSE_VISUAL_PROGRESS: CourseWaffleFlag(
+            namespace,
+            ENABLE_COURSE_VISUAL_PROGRESS,
+            flag_undefined_default=False
+        )
+    }
+
+
+def visual_progress_enabled(course_key):
+    """
+    Exposes varia of visual progress feature.
+        ENABLE_COMPLETION_TRACKING, current_site.configuration, AND
+        enable_course_visual_progress OR enable_visual_progress
+
+    :return:
+
+        bool -> True if site/course/global enabled for visual progress tracking
+    """
+    if not waffle().is_enabled(ENABLE_COMPLETION_TRACKING):
+        return
```

Comment:
(Warning, what I'm about to say is pedantic.)

The name of this method suggests that it will return a boolean value.  But all of these bare returns mean it will return either `None`, `True`, or `False`.  This doesn't matter much functionally, since `None` is False, but just to reduce any possible confusion, it would be better to explicitly return `False` in all of the places that you have a bare `return`.

Reply:
Fair enough. I'm flying a little blind with this waffle stuff, so I'll make it a little prettier.